bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlValue.java

```diff
@@ -186,6 +186,15 @@ protected void addDescendants(List<BXml> descendants, XmlItem currentElement, Li
         }
     }
 
+    protected void addDescendants(List<BXml> descendants, XmlItem currentElement) {
+        for (BXml child : currentElement.getChildrenSeq().getChildrenList()) {
+            if (child.getNodeType() == XmlNodeType.ELEMENT) {
```

Comment:
I think this should include non-elements too. Did you confirmed in the spec that this only include elements?

Reply:
Yes, the spec was rephrased to remove ambiguity. https://github.com/ballerina-platform/ballerina-spec/issues/701
It could include non-elements

Reply:
eg: 
```ballerina
<A>
   <B>
      <C>Hi</C>
   </B>
</A>
```

calling descendants on this results in the following sequence.
```ballerina
<B>
  <C>Hi</C>
</B>

<C>Hi</C>
```

Reply:
This test case feels wrong to me. https://github.com/ballerina-platform/ballerina-lang/pull/29302/files#diff-17b929237f4f118a94e336c2e1c226975c3ec2402973ec77a078689921db9cc7R530
The processing instruction is missing from the result

Reply:
This is because the processing instruction is an immediate child of `bs` which gets filtered as it's not an element.
If we had something like below where it's not an immediate child. It will be included in the sequence along with its parent element and then filtered out moving down the elements.
```ballerina
<bs><bk><?xml-stylesheet type=""text/xsl""?><t><en><!-- english --><txt>Everyday Italian</txt></en></t></bk></bs>
```

getDescendants() result:
```ballerina
<bk>
  <?xml-stylesheet type="text/xsl"?>
  <t>
    <en>
      <!-- english -->
      <txt>Everyday Italian</txt>
    </en>
  </t>
</bk>

<t>
  <en>
    <!-- english -->
    <txt>Everyday Italian</txt>
  </en>
</t>
<en>
  <!-- english -->
  <txt>Everyday Italian</txt>
</en>
<txt>Everyday Italian</txt>
```

Reply:
Function documentation says `The descendants of an element are the children of the element together with, for each of those children that is an element, ...`
Here the (in the test) we need to take all the children of `bk` + apply the same function to each element child.
So children of `bk` includes non elements.

Reply:
Updated in f1161288d594fxc24acddd7fdf86134030619d0fb